cura/OneAtATimeIterator.py

```diff
@@ -38,104 +41,50 @@ def _fillStack(self) -> None:
             self._node_stack = node_list[:]
             return
 
-        # Copy the list
-        self._original_node_list = node_list[:]
-
-        # Initialise the hit map (pre-compute all hits between all objects)
-        self._hit_map = [[self._checkHit(i, j) for i in node_list] for j in node_list]
+        hit_checker = HitChecker(node_list)
 
-        # Check if we have to files that block each other. If this is the case, there is no solution!
-        for a in range(0, len(node_list)):
-            for b in range(0, len(node_list)):
-                if a != b and self._hit_map[a][b] and self._hit_map[b][a]:
-                    return
+        if PrintOrderManager.isUserDefinedPrintOrderEnabled():
+            self._node_stack = self._getNodesOrderedByUser(hit_checker, node_list)
+        else:
+            self._node_stack = self._getNodesOrderedAutomatically(hit_checker, node_list)
+
+            # update print orders so that user can try to arrange the nodes automatically first
+            # and if result is not satisfactory he/she can switch to manual mode and change it
+            for index, node in enumerate(self._node_stack):
+                node.printOrder = index + 1
+
+    @staticmethod
+    def _getNodesOrderedByUser(hit_checker: HitChecker, node_list: List[CuraSceneNode]) -> List[CuraSceneNode]:
+        nodes_ordered_by_user = sorted(node_list, key=lambda n: n.printOrder)
+        if hit_checker.canPrintNodesInProvidedOrder(nodes_ordered_by_user):
+            return nodes_ordered_by_user
+        return []  # No solution
+
+    @staticmethod
+    def _getNodesOrderedAutomatically(hit_checker: HitChecker, node_list: List[CuraSceneNode]) -> List[CuraSceneNode]:
+        # Check if we have two files that block each other. If this is the case, there is no solution!
+        if hit_checker.anyTwoNodesBlockEachOther(node_list):
+            return []  # No solution
 
         # Sort the original list so that items that block the most other objects are at the beginning.
         # This does not decrease the worst case running time, but should improve it in most cases.
-        sorted(node_list, key = cmp_to_key(self._calculateScore))
+        sorted(node_list, key = cmp_to_key(hit_checker.calculateScore))
```

Comment:
looks like need to save the result here:
`node_list = sorted(node_list, key = cmp_to_key(hit_checker.calculateScore))`